Model/Indexer/SyncQueueRunner.php

```diff
@@ -62,7 +62,10 @@ public function executeFull()
                 try {
                     $job = Magento\Framework\App\ObjectManager::getInstance()->get($jobClass);
                     $jobResponse = $job->execute($data, $storeId);
-                    if ($jobResponse === true) {
+                    if ($jobResponse === 'back_in_queue') {
```

Comment:
Use a static const in `QueueManager`  -> called `JOB_EXECUTION_BACK_IN_QUEUE` 
Use it here and in Services/Queue/Processors/IndexProductsProcessor.php